doc/src/build/rest-api.md

```diff
@@ -246,7 +295,80 @@ NOTE: A Publish endpoint is in the works, but for now the only way to add a new
 
 To learn more about what `args` are accepted in a Move call, refer to the [SuiJSON](sui-json.md) documentation.
 
-#### POST /sync
+### POST /publish
+
+Publish Move module. 
+
+```shell
+curl --location --request POST $SUI_GATEWAY_HOST/publish \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "sender": "{{owner_address}}",
+    "compiledModules": {{vector_of_compiled_modules}},
+    "gasObjectId": "{{gas_object_id}}",
+    "gasBudget": 10000
+}' | json_pp
+```
+
+This endpoint will perform proper verification and linking to make
+sure the package is valid. If some modules have initializers, these initializers
+will also be executed in Move (which means new Move objects can be created in
+the process of publishing a Move package). Gas budget is required because of the
+need to execute module initializers.
+
+You should replace `{{owner_address}}` in the
+command above with an actual address value, for example one obtained
+from [`GET /addresses`](#get-addresses). You should also replace `{{gas_object_id}}` in the command above with an actual object ID, for example one obtained from [`GET
+/objects`](#get-objects) (from `objType` in the output of [`GET
+/objects`](#get-objects). You can see that all objects generated
+during genesis are of `Coin/SUI` type). For this call to work, `{{gas_object_id}}` must
+be owned by the address represented by `{{owner_address}}`.
+
+To publish a Move module you also need `{{vector_of_compiled_modules}}`. To generate the value of this field you can use the `sui-move` command showcased in the [Move documentation](move.md#publishing-a-package). i.e. 
+
+You can copy the ouputting hex representation of the compiled Move module into the 
```

Comment:
I am not sure I get this, particularly part about "generating object ID". Here's a suggestion for the rewrite (if I understood things correctly):

"Below you can see a sample output of `POST /publish`. One of the results of executing this command is generation of a package object representing the published Move code. An ID of the package object can be used as an argument for subsequent [Move calls](post-call) to functions defined in this package.